### ISSUE TITLE
Multi-Class Classification. Alternative for torch.stack. Finding it expensive and time consuming while training.

### ISSUE THREAD

**Issue by grimreaper94** [bentrevett/pytorch-sentiment-analysis#20]
CNN(
  (embedding): Embedding(82672, 50)
  (convs): ModuleList(
    (0): Conv2d(1, 50, kernel_size=(2, 50), stride=(1, 1), padding=(1, 0))
    (1): Conv2d(1, 50, kernel_size=(3, 50), stride=(1, 1), padding=(2, 0))
    (2): Conv2d(1, 50, kernel_size=(4, 50), stride=(1, 1), padding=(3, 0))
    (3): Conv2d(1, 50, kernel_size=(5, 50), stride=(1, 1), padding=(4, 0))
  )
  (fc): Linear(in_features=200, out_features=13, bias=True)
  (dropout): Dropout(p=0.5)
)

I have about 13 target fields. During training, I'm passing the batch data with <code>model(batch.data)</code>where<code>data</code> is my text field. But while calculating the loss, I'm trying to stack 13 one hot encoded vectors with <code>torch.stack([batch.target1, batch.target2.........,batch.target13], dim=1)</code>.  Is there an alternative to pass all my target field of the particular batch to have a faster computation?
My loss is BCEwithLogitsLoss. I am passing the out of fully connected layer to this function and doing a back prop.

**Comment by bentrevett** [comment]
Each of your examples is one of thirteen classes? Or can it be multiple of the thirteen?

**Comment by grimreaper94** [comment]
One of thirteen classes.

**Comment by bentrevett** [comment]
OK, I've added a notebook to the main tutorial. This uses the built in TREC datatset (which technically isn't sentiment analysis) that has 6 classes.

https://github.com/bentrevett/pytorch-sentiment-analysis/blob/master/5%20-%20Multi-class%20Sentiment%20Analysis.ipynb
